wlauto/workloads/iozone/__init__.py

```diff
@@ -0,0 +1,228 @@
+#    Copyright 2012-2015 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from wlauto import Workload, Parameter, Executable
+from wlauto.exceptions import ConfigError
+from wlauto.utils.types import list_of_strs
+import os
+import re
+import csv
+
+iozone_results_txt = 'iozone_results.txt'
+
+
+class Iozone(Workload):
+    name = 'iozone'
+    description = """
+    Iozone is a filesystem benchmark that runs a series of disk
+    I/O performance tests.
+
+    By default, iozone will run all tests in auto mode. To run
+    specific tests, they must be written in the form of:
+
+    ['0', '1', '5', '6']
+
+    The official website for iozone is at www.iozone.org.
+    """
+
+    parameters = [
+        Parameter('tests', kind=list_of_strs, default=[],
```

Comment:
- Why `list_of_strs`? Seems that `list_of_ints` would be more appropriate.
- Should set `allowed_values=range(13)` as only 0-12 indicate valid tests
- `default` can be omitted